c7n/resources/aws.py

```diff
@@ -157,33 +206,29 @@ def get_bucket_region_clientless(bucket, s3_endpoint):
     bucket_endpoint = f'https://{s3_endpoint_parts.netloc}/{bucket}'
     request = Request(bucket_endpoint, method='HEAD')
     try:
+        # Unauthenticated request always raise an error the status code
```

Comment:
We expect/hope this to be true - the unauthenticated request only succeeds against public buckets. I wonder if we should actually bomb or yell if we _don't_ get an HTTP error here, since it implies you're writing custodian logs to a spot with a questionable access policy :thinking:

Reply:
cool, I was wondering since the success case wasn't in tests. I've gone ahead and made it so if we detect a public bucket we default to raise an invalid output error.

Reply:
its worth noting this is not a hard boundary, ie we detect and warn on public if we're in the same region, but a region redirect to a public bucket, would need a subsequent request within the region  to verify on the bucket's access nature.